components/script/dom/bindings/utils.rs

```diff
@@ -112,6 +115,7 @@ unsafe impl Sync for DOMClass {}
 
 /// The JSClass used for DOM object reflectors.
 #[derive(Copy)]
+#[repr(C)]
```

Comment:
This is a tiny addition with a big impact. Adding the `depth` field to `DOMClass` caused rustc to reorder the fields of `DOMJSClass`, which broke lots of code that assumed that casting a `JSClass` pointer to `DOMJSClass` would yield a valid object.